auth_service/user_management/claims_repository/core/claims.py

```diff
@@ -37,17 +43,40 @@ def is_valid_claim(claim: Claim, now: Callable[[], DateTimeUTC] = now_as_utc) ->
     return not claim.revocation_date and claim.valid_from <= now() <= claim.valid_until
 
 
+def is_internal_claim(claim: Claim, visa_type: VisaType) -> bool:
+    """Check whether this is a valid internal claim of the given type."""
+    return (
+        claim.visa_type == visa_type
+        and claim.source == INTERNAL_SOURCE
+        and claim.asserted_by not in (None, AuthorityLevel.SELF)
+        # conditions are currently not supported,
```

Comment:
Expiration is checked separately in `is_valid_claim` so that it can be reused elsewhere.

Do you think this should be called in all specific checks automatically? Otherwise I would just mention in the document that this check is not included.

Reply:
OK, I think that is fine.